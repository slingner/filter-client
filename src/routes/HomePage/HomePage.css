.wrapper {
  display: flex;
  align-items: stretch;
  min-height: 100vh;
}

.menu-toggle {
  display: none;
}

/* To hide the checkbox */
#hamburger-menu-toggle-1 {
  display: none;
}

/* Styles for the 'open' state, if the checkbox is checked */
#hamburger-menu-toggle-1:checked ~ .sidebar {
  display: initial;
  /* see Step 3 for the rest of the styles */
}


@media (max-width: 767px) {
  .sidebar {
    display: none;
  }
  .menu-toggle {
    display: initial; /* display the toggle, previously hidden */
  }
}