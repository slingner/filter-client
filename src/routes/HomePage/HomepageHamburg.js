import React, { Component } from 'react';

class HamburgerMenuWrapper extends Component {
  constructor(props) {
    super(props);
    this.state = { isOpen: false };
  }

  toggleMenu() {
    this.setState({ isOpen: !this.state.isOpen });
  }

  render() {
    return (
      <div class="wrapper">
        {this.state.isOpen && <div class="sidebar">sidebar</div>}
        <button onClick={() => this.toggleMenu()}>Toggle Menu</button>
        <div class="main-content">content</div>
      </div>
    );
  }
}

export default HamburgerMenuWrapper