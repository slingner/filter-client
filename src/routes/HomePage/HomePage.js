import React, { Component } from 'react';
// import coffee4 from './Assets/coffee4.jpg';
// import coffee3 from './Assets/coffee3.jpg';
// import coffee2 from './Assets/coffee2.jpg';
import beanlist from './Assets/beanlist.png';
import savedbeans from './Assets/savedbeans.png';
import './HomePage.css';
import { NavLink } from 'react-router-dom';
import TokenService from '../../services/token-service';
import { ReactComponent as FilterLogo } from './Assets/ICON.svg';

import { ReactComponent as CoffeeBirdy } from './Assets/coffeebirdy.svg';

export default class HomePage extends Component {
  render() {
    return (
      <>
        <section className="bean-in-hand-img">
          <FilterLogo className="homeLogo" />

          <p className="headertext">
            Discover and explore small batch roasted coffee beans from around
            the world by filtering through a collection of beans by their unique
            flavor profiles.
          </p>
        </section>
        <div className="HomePage-wrapper">
          <div className="content">
            <header>
              <h3>Keep track and learn about your favorite beans</h3>
            </header>
            <p>
              {' '}
              Browse through the collection to explore each bean's Country of
              Origin, MASL(Meters Above Sea Level), Grower, Process and Flavor
              Notes.{' '}
            </p>

            <p>
              Once you've found some beans that give you a jolt, save them to
              your personalized collection and leave notes for yourself
              regarding each bean.
            </p>

            <p>
              FILTER gives you the chance to explore coffee that is responsibly
              sourced from crops around the world.{' '}
            </p>
          </div>
          <br />
          {TokenService.hasAuthToken() ? (
            ''
          ) : (
            <NavLink to="/register" className="signUp">
              Get Started
            </NavLink>
          )}
          <CoffeeBirdy
            style={{
              fill: 'rgb(130, 114, 93)',
              width: '150px',
              margin: 'auto',
              marginTop: '40px',
            }}
          />
          {/* <p>
            Demo FILTER and see how to save and take notes on each card by{' '}
            <NavLink className="Links" to="/login">
              logging in
            </NavLink>{' '}
            using:{' '}
          </p>
          <p> Username: dunder</p>
          <p> Password: password</p>
          <p></p> */}
        </div>
      </>
    );
  }
}
