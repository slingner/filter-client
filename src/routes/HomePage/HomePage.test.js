import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import HomePage from './HomePage'


describe(`<HomePage />`, () => {
  //Snapshot Testing
  it('renders HomePage by default', () => {
    const wrapper = shallow(<HomePage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})