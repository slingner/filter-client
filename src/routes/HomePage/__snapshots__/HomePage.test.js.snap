// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders HomePage by default 1`] = `
<Fragment>
  <section
    className="bean-in-hand-img"
  >
    <ForwardRef(SvgIcon)
      className="homeLogo"
    />
    <p
      className="headertext"
    >
      Discover and explore small batch roasted coffee beans from around the world by filtering through a collection of beans by their unique flavor profiles.
    </p>
  </section>
  <div
    className="HomePage-wrapper"
  >
    <div
      className="content"
    >
      <header>
        <h3>
          Keep track and learn about your favorite beans
        </h3>
      </header>
      <p>
         
        Browse through the collection to explore each bean's Country of Origin, MASL(Meters Above Sea Level), Grower, Process and Flavor Notes.
         
      </p>
      <img
        alt="savedbeans"
        className="screenshot-saved-bean-img"
        src="beanlist.png"
      />
      <p>
        Once you've found some beans that give you a jolt, save them to your personalized collection and leave notes for yourself regarding each bean!
      </p>
      <img
        alt="savedbeans"
        className="screenshot-saved-bean-img"
        src="savedbeans.png"
      />
      <p>
        FILTER gives you the chance to explore coffee that is responsibly sourced from crops around the world.
         
      </p>
    </div>
    <br />
    <NavLink
      to="/register"
    >
      Sign Up Now!
    </NavLink>
    <ForwardRef(SvgCoffeebirdy)
      style={
        Object {
          "fill": "rgb(130, 114, 93)",
          "margin": "auto",
          "marginTop": "30px",
          "width": "150px",
        }
      }
    />
  </div>
</Fragment>
`;
